### PR TITLE
1608 download filtered results UI modal

### PR DESCRIPTION
**Pull request by sergioloporto**
#1608 

**Changes**
* User has the ability to download full dataset that is stored in S3 and updated daily (for now only in .csv)
* User has the ability to download filtered data instantly if applied query returns less than 10000 cases (csv, tsv and json)
* For bigger queries background worker will fetch the results and after compressing it upload to AWS S3 to newly created `covid19-filtered-downloads` bucket. The file will be automatically removed after 3 days. The user will receive an email containing pre signed url to this file (also valid for 3 days) allowing him to download the results. (csv, tsv and json)

Email that is being sent with the download link
Screen Shot 2021-06-01 at 11 47 12

Alert on data page informing user that the email was sent
Screen Shot 2021-06-03 at 10 37 48

### CODE REVIEW

File: data-serving/data-service/src/controllers/case.ts [hunk 1]
```diff
@@ -74,6 +74,13 @@ export class CasesController {
      * Handles HTTP POST /api/cases/download.
      */
     download = async (req: Request, res: Response): Promise<void> => {
+        if (req.body.query && req.body.caseIds) {
+            res.status(400).json({ message: 'Bad request' });
+            return;
+        }
```
Comment (iamleeg) [inline data-serving/data-service/src/controllers/case.ts:80]: Nice, good to test this constraint before continuing

File: data-serving/data-service/src/controllers/case.ts [hunk 2]
```diff
@@ -105,39 +112,105 @@ export class CasesController {
...
+                    })
+                    .limit(queryLimit);
+            } else {
+                matchingCases = await Case.aggregate(
+                    casesIgnoringExcluded,
+                ).collation({
+                    locale: 'en_US',
+                    strength: 2,
+                });
+            }
```
Comment (iamleeg) [inline data-serving/data-service/src/controllers/case.ts:131]: Not a big deal, but could be changed to:

```
let query:any = Case.aggregate(casesIgnoringExcluded)collation({locale: 'en_US', strength: 2, });
if (queryLimit) { query = query.limit(queryLimit); }
const matchingCases = await query;
```

File: verification/curator-service/api/src/workers/downloadAsync.js [hunk 3]
```diff
@@ -14,64 +17,130 @@ try {
...
+
+                    resolve(url);
+                });
+            });
+
+            // Send signed url via email to user
+            const mailMessage = `
+            <p>Please visit the link below to download list of cases in response to your query <strong>${workerData.query}</strong>.</p>
+
+            <a href="${signedUrl}">Click here to download the data</a>
```
Comment (iamleeg) [inline verification/curator-service/api/src/workers/downloadAsync.js:104]: Nice, a much better way of handling the large files

**Review by iamleeg — APPROVED** [review]
Nice work, thank you!

File: data-serving/data-service/src/controllers/case.ts (hunk 2 above)
Comment (maciekz1996) [reply data-serving/data-service/src/controllers/case.ts:131]: Yeah, the code will be cleaner, thank you for spotting that!

**Pull request merged**